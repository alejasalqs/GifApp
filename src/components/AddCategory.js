import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState(''); // Estado inicial

    // Esta función se ejcuta cuando escribimos
    const handleInputChange = ( e ) => {
        setInputValue(e.target.value)
    }

    // esta función se ejcuta al presionar enter en el input
    const handleSubmit = (e) => {
        e.preventDefault();
        // Llamamos la referencia de la funcion que esta en el otro componente
        if(inputValue.trim().length >= 2) {
           setCategories(categories => [inputValue]);
           setInputValue('');
        } else {
            alert('This is not a valid search');
        }
    } 

    return (
        <form onSubmit={handleSubmit}>
            <input 
                type="text" 
                value={inputValue}
                onChange={handleInputChange}
            ></input>
        </form>
    )
}

// Props obligatorios
AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}