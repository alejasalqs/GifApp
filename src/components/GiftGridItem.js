import React from 'react'
import PropTypes from 'prop-types';


export const GiftGridItem = ( { title, url} ) => {

    return (
        <div className="card animate__animated animate__fadeIn">
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
    )
}

// Props obligatorios
GiftGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
