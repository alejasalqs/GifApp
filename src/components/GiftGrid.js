import React from 'react'
import { useFetchGifts } from '../hooks/useFetchGifts';
import { GiftGridItem } from './GiftGridItem';
import PropTypes from 'prop-types';

export const GiftGrid = ({ category }) => {

    // La l√≥gica para obtener las imagenes esta en este custom hook
    // Se renombra la varibale data por gifts
    const { data:gifts, loading } = useFetchGifts(category);

    return (
        <>
            { loading && <h1>Loading...</h1>}
            <h3>Searching: {category}</h3>
            <div className="card-grid">
                {
                    gifts.map(gift => (
                        <GiftGridItem 
                        key={gift.id} 
                        {...gift} />))
                }
            </div>
        </>
    )
}

GiftGrid.propTypes = {
    category: PropTypes.string.isRequired
}