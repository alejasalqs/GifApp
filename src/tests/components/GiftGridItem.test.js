import React from 'react';
import { shallow } from 'enzyme';
import { GiftGridItem } from '../../components/GiftGridItem'; 

describe('Pruebas en el componente GifGridItem', () => {

    const title = 'Hello test';

    const url = 'https://localhost/test.jpg';

    const wrapper = shallow( <GiftGridItem title={title} url={url}/> );

    test('Debe mostrar el componente correctamente ', () => {

        expect( wrapper ).toMatchSnapshot();
        
    })
    
    test('Debe mostrar un parrafo con el title ', () => {
        const p = wrapper.find('p');

        expect( p.text().trim() ).toBe(title);
    })
    
    test('Debe de tener la imagen igual al los props y el alt igual al title ', () => {
        const img = wrapper.find('img');

        expect(img.prop('src')).toBe(url);

        expect(img.prop('alt')).toBe(title);
    })
    
    test('Debe tener la clase fadeIn ', () => {
        const div = wrapper.find('div');

        expect(div.prop('className').includes('fadeIn')).toBe(true);
    })
    
})
