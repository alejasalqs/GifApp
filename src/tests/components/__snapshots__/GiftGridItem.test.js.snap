// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGridItem Debe mostrar el componente correctamente  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Hello test"
    src="https://localhost/test.jpg"
  />
  <p>
    Hello test
  </p>
</div>
`;
